From: Ben Hutchings <ben@decadent.org.uk>
Date: Sun, 28 Jul 2013 05:52:40 +0100
Subject: perf: Avoid ABI change in 3.2.49

Some mmap metadata was moved from struct perf_event to struct
ring_buffer.  These fields should never have been used by OOT modules,
but need to be retained so the following field offsets don't change.

--- a/include/linux/perf_event.h
+++ b/include/linux/perf_event.h
@@ -823,6 +823,9 @@ struct perf_event {
 	/* mmap bits */
 	struct mutex			mmap_mutex;
 	atomic_t			mmap_count;
+	/* These two are unused but kept here for ABI compat */
+	int				mmap_locked;
+	struct user_struct		*mmap_user;
 
 	struct ring_buffer		*rb;
 	struct list_head		rb_entry;
